### PR TITLE
Move card to tab - analytics event

### LINKED ISSUE #35123: Snowplow: Add new event for card moved to different tab
## Context
We want to see how people are moving cards between tabs. Add a new event `card_moved_to_tab` to the `dashboard` Snowplow schema. 

[More details here](https://www.notion.so/metabase/Card-moved-to-different-tab-eb02a47dc27c4338bb025d4812b3f3d8?pvs=4).

### PR DESCRIPTION
**Pull request by npretto**
### Description

this closes #35123 by adding the relevant event

### How to verify

E2e should pass

### CODE REVIEW

File: snowplow/iglu-client-embedded/schemas/com.metabase/dashboard/jsonschema/1-1-2 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        "auto_apply_filters_disabled",
+        "dashboard_tab_created",
+        "dashboard_tab_deleted",
+        "new_text_card_created",
+        "new_heading_card_created",
+        "new_link_card_created",
+        "new_action_card_created",
+        "card_set_to_hide_when_no_results",
+        "dashboard_pdf_exported",
+        "card_moved_to_tab"
```
Comment (npretto) [inline snowplow/iglu-client-embedded/schemas/com.metabase/dashboard/jsonschema/1-1-2:27]: this is the new event

File: frontend/src/metabase/dashboard/analytics.ts [hunk 2]
```diff
@@ -26,3 +26,10 @@ export const trackCardCreated = (type: CardTypes, dashboard_id: number) => {
     dashboard_id,
   });
 };
+
+export const trackCardMoved = (dashboardId: DashboardId) => {
+  trackSchemaEvent("dashboard", "1-1-2", {
```
Comment (npretto) [inline frontend/src/metabase/dashboard/analytics.ts:31]: Here I used 1-1-2, should other events in this file be updated to that version too?

File: e2e/test/scenarios/dashboard/tabs.cy.spec.js [hunk 3]
```diff
@@ -363,4 +363,24 @@ describeWithSnowplow("scenarios > dashboard > tabs", () => {
     saveDashboard();
     expectGoodSnowplowEvents(PAGE_VIEW_EVENT + 2); // dashboard_tab_deleted
   });
+
+  it(
+    "should send snowplow events when cards are moved between tabs",
+    { scrollBehavior: false },
```
Comment (npretto) [inline e2e/test/scenarios/dashboard/tabs.cy.spec.js:369]: With the automatic scroll the "move tab" icon scrolls under the header, making the test fail.
Is there a better solution?

Comment (npretto) [reply e2e/test/scenarios/dashboard/tabs.cy.spec.js]: (I'm currently checking if it's true that it's flaky, as I couldn't find the failed test on deploy sentinel to link here)

File: frontend/src/metabase/dashboard/analytics.ts (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/dashboard/analytics.ts:31]: Yes

File: e2e/test/scenarios/dashboard/tabs.cy.spec.js [hunk 4]
```diff
@@ -363,4 +363,20 @@ describeWithSnowplow("scenarios > dashboard > tabs", () => {
...
+
+    editDashboard();
+    createNewTab();
+    goToTab("Tab 1");
+
+    expectGoodSnowplowEvents(PAGE_VIEW_EVENT);
+
+    moveDashCardToTab({ tabName: "Tab 2" });
+
+    expectGoodSnowplowEvents(PAGE_VIEW_EVENT + 1);
```
Comment (ranquild) [inline e2e/test/scenarios/dashboard/tabs.cy.spec.js:380]: Please another helper https://github.com/metabase/metabase/blob/92312aba6932c33f00aadd15b06763237098939b/e2e/test/scenarios/collections/uploads.cy.spec.js#L124, where the event name is specified

File: snowplow/iglu-client-embedded/schemas/com.metabase/dashboard/jsonschema/1-1-2 [hunk 5]
```diff
@@ -0,0 +1,70 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Dashboard events",
+  "self": {
+    "vendor": "com.metabase",
+    "name": "dashboard",
+    "format": "jsonschema",
+    "version": "1-1-1"
```
Comment (ranquild) [inline snowplow/iglu-client-embedded/schemas/com.metabase/dashboard/jsonschema/1-1-2:8]: Please update the version here

File: frontend/src/metabase/dashboard/components/DashCard/DashCardTabMenu/DashCardTabMenu.tsx [hunk 6]
```diff
@@ -39,8 +46,10 @@ export function DashCardTabMenu({ dashCardId }: DashCardTabMenuProps) {
           },
         }),
       );
+
+      trackCardMoved(checkNotNull(dashboardId));
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/components/DashCard/DashCardTabMenu/DashCardTabMenu.tsx:50]: Please move tracking to the redux action `moveDashCardToTab`

**Review by ranquild — COMMENTED** [review]
Left comments, please take a look

File: frontend/src/metabase/dashboard/components/DashCard/DashCardTabMenu/DashCardTabMenu.tsx (hunk 6 above)
Comment (npretto) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardTabMenu/DashCardTabMenu.tsx:50]: The action right now is just a plain action (no thunk)
There was a suggestion about converting it to a thunk to also include the undo toast, but it was reverted following the same discussion as there wasn't consensus about it https://github.com/metabase/metabase/pull/34961#discussion_r1371280900

Should I re-convert it to a thunk that includes the analytics event?

File: frontend/src/metabase/dashboard/analytics.ts (hunk 2 above)
Comment (npretto) [reply frontend/src/metabase/dashboard/analytics.ts]: Thanks, I extracted it to a constant

File: e2e/test/scenarios/dashboard/tabs.cy.spec.js [hunk 7]
```diff
@@ -363,4 +364,30 @@ describeWithSnowplow("scenarios > dashboard > tabs", () => {
     saveDashboard();
     expectGoodSnowplowEvents(PAGE_VIEW_EVENT + 2); // dashboard_tab_deleted
   });
+
+  it("should send snowplow events when cards are moved between tabs", () => {
+    const CARD_MOVE_EVENT_NAME = "card_moved_to_tab";
```
Comment (ranquild) [inline e2e/test/scenarios/dashboard/tabs.cy.spec.js:369]: I think we usually use regular camelCase for local constants

**Review by ranquild — APPROVED** [review]

**Pull request merged**